signer/xpi/cose.go

```diff
@@ -65,41 +66,47 @@ func intToCOSEAlg(i int) (v *cose.Algorithm) {
 func (s *XPISigner) generateCOSEKeyPair(coseAlg *cose.Algorithm) (eeKey crypto.PrivateKey, eePublicKey crypto.PublicKey, err error) {
 	var signer *cose.Signer
 
+	switch key := s.issuerKey.(type) {
+	case *rsa.PrivateKey, *ecdsa.PrivateKey, *crypto11.PKCS11PrivateKeyRSA, *crypto11.PKCS11PrivateKeyECDSA:
+		// ok
+	default:
+		err = errors.Errorf("xpi: Cannot generate COSEKeypair for issuerKey type %T", key)
+		return
+	}
+
 	switch coseAlg {
-	case nil:
-		err = errors.New("Cannot generate private key for nil cose Algorithm")
 	case cose.PS256:
 		var size = 2048
 		switch key := s.issuerKey.(type) {
-		case *ecdsa.PrivateKey:
-			if key.D.BitLen() > size {
-				size = key.D.BitLen()
```

Comment:
This is never true for the common NIST curves defined in `crypto/elliptic` since the ECDSA private key / D param bit len will always be much less than 2048: https://play.golang.org/p/_DrG2-g2ZxF

It doesn't make sense to use this directly since it's 

It's also a weird case since we're generating an RSA EE for an ECDSA issuer.

Instead of dropping this case and just using the default like we do here we could:

a) try to compute the effective RSA key size for a give ECDSA key size and generate a key of that size
b) bump the default up to 4096 (but this would hurt perf if we start signing with a bunch of PS256)
c) make the default RSA key size configurable at a signer or mode level (e.g. webext's use 2048, but system addons use 4096)
d) keep the above case which is effectively using the default size

Reply:
per chat with ulfr: we can worry about it later is we start using PS256 sigs.